python/lsst/ts/MTDomeTrajectory/mock_dome.py

```diff
@@ -0,0 +1,278 @@
+        motion_state : `lsst.ts.idl.Dome.MotionState`
+            Motion state at end of move.
+        """
+        end_tai = self.azimuth_actuator.path.segments[-1].tai
+        duration = end_tai - salobj.current_tai()
+        if duration > 0:
+            await asyncio.sleep(duration)
+        self.evt_azMotion.set_put(
+            state=motion_state, inPosition=in_position,
+        )
+
+    async def report_elevation_done(self, in_position, motion_state):
+        """Wait for elevation to stop moving and report it in position.
+
+        Parameters
+        ----------
+        in_position : `bool`
+            Is the axis in position at the end of the move?
+        motion_state : `lsst.ts.idl.Dome.MotionState`
+            Motion state at end of move.
+        """
+        duration = self.elevation_actuator.remaining_time()
+        if duration > 0:
+            await asyncio.sleep(duration)
+        self.evt_elMotion.set_put(
+            state=motion_state, inPosition=in_position,
+        )
+
+    def stop_azimuth(self):
+        """Stop the azimuth actuator and the done task.
+
+        Report not in position, if changed.
+
+        Unlike do_stopAz this does not require that the CSC is enabled.
+        """
+        self.azimuth_actuator.stop()
+        if self.azimuth_done_task.done():
+            return
+        self.azimuth_done_task.cancel()
+        self.evt_azMotion.set_put(
+            state=Dome.MotionState.STOPPING, inPosition=False,
+        )
+        self.azimuth_done_task = asyncio.create_task(
+            self.report_azimuth_done(
+                in_position=False, motion_state=Dome.MotionState.STOPPED
+            )
+        )
+
+    def stop_elevation(self):
+        """Stop the elevation actuator and the done task.
+
+        Report not in position, if changed.
+
+        Unlike do_stopEl this does not require that the CSC is enabled.
+        """
+        self.elevation_actuator.stop()
+        if self.elevation_done_task.done():
+            return
+        self.elevation_done_task.cancel()
+        self.evt_elMotion.set_put(
+            state=Dome.MotionState.STOPPING, inPosition=False,
+        )
+        self.evt_elMotion.set_put(
+            state=Dome.MotionState.STOPPED, inPosition=False,
+        )
+
+    async def telemetry_loop(self):
+        try:
+            while True:
+                tai = salobj.current_tai()
+                azimuth_target = self.azimuth_actuator.target.at(tai)
+                azimuth_actual = self.azimuth_actuator.path.at(tai)
+                self.tel_azimuth.set_put(
+                    positionActual=azimuth_actual.position,
+                    positionCommanded=azimuth_target.position,
+                )
+
+                self.tel_lightWindScreen.set_put(
+                    positionActual=self.elevation_actuator.position(tai),
+                    positionCommanded=self.elevation_actuator.end_position,
+                )
```

Comment:
The elevation actuator (a PointToPoint actuator) is much simpler than the elevation actuator (a TrackingActautor), so it is easier to get the information I want directly. For a TrackingActuator I have to call the at method.

For the current code it doesn't really matter, but it will help when the Dome telemetry is enhanced to allow reporting dome azimuth.